### PR TITLE
docs(grid): add components playgrounds

### PR DESCRIPTION
**Pull request by brandyscarney**
This PR adds the following component playgrounds:

- Basic Usage
- Fixed Grid
- Column Size
  - Content-based size
  - Specified size
  - Responsive size
- Column Offset
  - Specified offset
  - Responsive offset
- Column Push & Pull
  - Specified push & pull
  - Responsive push & pull
- Alignment
  - Vertical Alignment
  - Horizontal Alignment
- Customizing the Grid
  - Fixed Width
  - Number of Columns
  - Padding

It updates the column & row API documentation to point to the grid API documentation.

⚠️ It also removes the grid documentation from layout here: https://ionicframework.com/docs/layout/grid
I searched and replaced any links to it. Added a redirect here: https://github.com/ionic-team/ionic-nginx/pull/20

Preview: https://ionic-docs-git-fw-1292-ionic1.vercel.app/docs/api/grid

### CODE REVIEW

**Review by amandaejohnston — CHANGES_REQUESTED** [review]
Love your examples, everything looks really snazzy 🎉

File: docs/layout/grid.md [hunk 1]
```diff
@@ -1,512 +0,0 @@
----
-title: Responsive Grid
```
[the remark this answers is not in the document]
Comment (brandyscarney) [reply docs/layout/grid.md:2]: Good call! Added a PR to redirect here: https://github.com/ionic-team/ionic-nginx/pull/20

**Comment by brandyscarney** [comment]
@liamdebeasi @sean-perkins Are you both okay with getting rid of the file https://ionicframework.com/docs/layout/grid completely and redirecting to the api grid documentation?

**Review by amandaejohnston — APPROVED** [review]

**Comment by liamdebeasi** [comment]
Agree on getting rid of the docs page in favor of the Grid component. We should not break the URL though, so a 301 redirect to the component page is an option here.

**Comment by brandyscarney** [comment]
@liamdebeasi Yeah there is a redirect added here: https://github.com/ionic-team/ionic-nginx/pull/20

@sean-perkins Added it back to the sidebar but it directs to `api/grid` here: https://github.com/ionic-team/ionic-docs/pull/2596/files#diff-9a6a0fdafefb0eecffb77784bc94f5ad984062167b8bcb7c4a899c90f6f567e8R33

Can be seen here: https://ionic-docs-git-fw-1292-ionic1.vercel.app/docs/

Let me know if you want any other changes!

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**